lib/rspec/core/formatters.rb

```diff
@@ -170,6 +170,11 @@ def add(formatter_to_use, *paths)
       end
     end
 
+    # @private
+    def change_output_stream(output_stream)
+      @formatters.each { |formatter| formatter.change_output(output_stream) }
```

Comment:
Well I guess we'd only be mutating state in one place then, seems sensible to me!